### PR TITLE
API Design Guidelines chapter 11: 'info' object, 'servers' object and cleanup 

### LINKED ISSUE #199: Remove basePath in servers.url as variable
**Problem description**
I noticed that we are misusing variables in servers.url properties in most API definitions.

Commonly we define `servers` as, e.g.:

```
servers:
  - url: "{apiRoot}/{basePath}"
    variables:
      apiRoot:
        default: http://localhost:9091
        description: API root
      basePath:
        default: qod/v0
        description: Base path for the QoD API
```

But `basePath` which is the api-name and major version is not something that may be changed by the client. `default` is defined in OAS as:

| Field Name | Type | Description |
| --|:---:|---|
| default	| string |	**REQUIRED**. The default value to use for substitution, and to send, if an alternate value is not supplied. Unlike the Schema Object's `default`, this value MUST be provided by the consumer. |

We are not allowing an alternate value to be supplied by the API client, so this part of the `servers.url` must be fixed. 

**Possible evolution**
Proposal is to redefine `servers` as, e.g.:

```
servers:
  - url: "{apiRoot}/qod/v0"
    variables:
      apiRoot:
        default: http://localhost:9091
        description: API root
```

### LINKED ISSUE #201: Add guidance for Info object in apis
[OpenAPI Info Object](https://spec.openapis.org/oas/v3.0.3#info-object)

Provide guidance on the info-object.  For instance the following fields could have consistency.

```yaml
  termsOfService: http://swagger.io/terms/
  contact:
    email: project-email@sample.com
```

### PR DESCRIPTION
**Pull request by rartych**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:

The definitions of 'info' and 'servers` object were added to chapter 11.
Other edits on API definition guidelines.
I propose to remove the figures as they do not add value to the text.

#### Which issue(s) this PR fixes:

Fixes #199 
Fixes #201 

#### Special notes for reviewers:

Info object described in the same manner as here: https://swagger.io/docs/specification/api-general-info/

#### Changelog input

```
The definitions of 'info' and 'servers` object were added to chapter 11

```

### CODE REVIEW

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -1068,20 +1068,54 @@ Below considerations should be checked when an API is documented:
...
+    This API allows to verify that the provided mobile phone number is the one used in the device. It
+    verifies that the user is using a device with the same mobile phone number as it is declared.
+    ### Authorization and authentication
+    CAMARA guidelines defines a set of authorization flows ...
+  # API version - Aligned to SemVer 2.0 according to CAMARA versioning guidelines
+  version: 1.0.1
+  # Link to the page that describes the terms of service - to be replaced by the provider' terms'
+  termsOfService: http://example.com/terms/
+  # Contact information: name, email, URL
+  contact:
```
Comment (shilpa-padgaonkar) [inline documentation/API-design-guidelines.md:1098]:
```suggestion
  contact:
  # Link to the provider's support page.
```

**Review by shilpa-padgaonkar — DISMISSED** [review]
Added a small comment to be consistent. Rest /LGTM

**Comment by shilpa-padgaonkar** [comment]
@patrice-conil , @PedroDiez and @RubenBG7 : Could you kindly review & approve this if all looks ok? we are trying to speed up reviews of at least the straightforward/simple PRs so that we are in time for the 0.4 release. Thanks in advance.

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by patrice-conil — APPROVED** [review]
/LGTM

**Review by shilpa-padgaonkar — DISMISSED** [review]
/LGTM

**Comment by shilpa-padgaonkar** [comment]
@PedroDiez or @RubenBG7  Could one of you kindly approve this PR so that we can proceed to merge? Thanks in advance.

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -1068,20 +1068,55 @@ Below considerations should be checked when an API is documented:
...
+    verifies that the user is using a device with the same mobile phone number as it is declared.
+    ### Authorization and authentication
+    CAMARA guidelines defines a set of authorization flows ...
+  # API version - Aligned to SemVer 2.0 according to CAMARA versioning guidelines
+  version: 1.0.1
+  # Link to the page that describes the terms of service - to be replaced by the provider's terms
+  termsOfService: http://example.com/terms/
+  # Contact information: name, email, URL
+  contact:
+    name: API Support
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1099]: One doubt 
which values are expected to be used in the APIs (I assume we will have to insert this information in every API so i think we could have some alignment regarding the value to be indicated to use the same approach among APIs or it is up to the WG the value to be provided).
Anycase, all of them (name, email, url) has to be indicated within the yaml, doesn`t?


name: <WG name> support -> e.g.  Blockchain Public Address  Support
email: mail distribution list of the WG -> e.g.  sp-bpa@lists.camaraproject.org
url: link to the WG site in CAMARA gtihub -> e.g. https://github.com/camaraproject/BlockchainPublicAddress

Comment (rartych) [reply documentation/API-design-guidelines.md:1099]: I think it is assumed that the contact information will be changed by each API Provider (Operator or Aggregator) when publishing API specification as the documentation for their service - possibly it should be explained in API Design Guidelines here in chapter 11.
When yaml is hosted in CAMARA Github all contact info is in main README.md
@tanjadegroot @hdamker WDYT?

**Comment by tanjadegroot** [comment]
Hi Rafal,

Yes that is indeed the assumption.
It is a good idea to have that I the example and guidelines.

Cheers
Tanja

File: documentation/API-design-guidelines.md (hunk 2 above)
Comment (jlurien) [reply documentation/API-design-guidelines.md:1099]: I think we should include the information that is relevant for CAMARA in our specs, as each API provider will adapt our OAS spec to whatever template or format they use in their portals. `title` and  `version` are the only required sub-sections for `info`  section. If we decided to add the `contact` sub-section, I would fill it with contact details for CAMARA.

Comment (shilpa-padgaonkar) [reply documentation/API-design-guidelines.md:1099]: I had added such contact subsection with Camara relevant details in the [common.yaml ](https://github.com/camaraproject/Commonalities/blob/main/artifacts/CAMARA_common.yaml )as a part of the last release and would personally prefer this format as well.

image

Comment (rartych) [reply documentation/API-design-guidelines.md]: Following the discussion in Release Management: https://wiki.camaraproject.org/display/CAM/2024-06-04+Release+WG+Minutes and https://github.com/camaraproject/Commonalities/pull/214#discussion_r1626072362 I have removed `contact` and `termsOfService` from the info object and added some explanation.

See the changes: https://github.com/camaraproject/Commonalities/pull/214/commits/2421afb69f4d59597a0aa10f6022e9387cc3483d.

**Comment by rartych** [comment]
Following the discussion in Release Management: https://wiki.camaraproject.org/display/CAM/2024-06-04+Release+WG+Minutes
I have added  `x-camara-commonalities` extension field to the info object.

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
/LGTM

**Review by hdamker — APPROVED** [review]
LGTM

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Pull request merged**